gnomad_constraint/utils/constraint.py

```diff
@@ -734,6 +732,67 @@ def calculate_mu_by_downsampling(
     return annotate_mutation_type(ht)
 
 
+def add_oe_lof_upper_rank_and_bin(
+    ht: hl.Table, use_mane_select_instead_of_canonical: bool = True
+) -> hl.Table:
+    """
+    Compute the rank and decile of the lof oe upper confidence interval for MANE Select or canonical ensembl transcripts.
+
+    :param ht: Input Table with the value for the lof oe upper confidence interval stored in ht.lof.oe_ci.upper.
+    :param use_mane_select_instead_of_canonical: Use MANE Select rather than canonical transcripts for filtering the Table.
+        If a gene does not have a MANE Select transcript, the canonical transcript (if available) will be used instead. Default is True.
+    :return: Table with anntotations added for 'upper_rank', 'upper_bin_decile'.
+    """
+    # Filter to only ensembl transcripts of the specified transcript filter. If MANE select is specified, and a gene
+    # does not have a MANE select transcript, use canonical instead.
+    if use_mane_select_instead_of_canonical:
+        genes = ht.group_by(ht.gene_id).aggregate(
+            mane_present=hl.agg.any(ht.mane_select),
+            canonical_present=hl.agg.any(ht.canonical),
+        )
+
+        genes = genes.annotate(
+            only_canonical=~(genes.mane_present) & (genes.canonical_present)
+        )
+
+        ms_ht = ht.annotate(
+            _only_canonical=genes[ht.gene_id].only_canonical,
+            _mane_present=genes[ht.gene_id].mane_present,
+        )
+        ms_ht = ms_ht.filter(
+            (ms_ht.transcript.startswith("ENST"))
+            & (
+                (ms_ht._mane_present & ms_ht.mane_select)
+                | (ms_ht._only_canonical & ms_ht.canonical)
+            )
+        )
+    else:
+        ms_ht = ht.filter((ht.canonical) & (ht.transcript.startswith("ENST")))
+
```

Comment:
it might be helpful to add a count here to check how many transcripts are removed (just in case something goes awry with the transcript filtering)